### PR TITLE
SONARIAC-571 Dangerous chmod options on RUN instructions

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

**Comment by rudy-regazzoni-sonarsource** [comment]
First version which handle Exec and Shell form for RunInstruction for chmod.
It does not support HeredocForm, will come in another PR.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/PosixPermissionCheck.java [hunk 1]
```diff
@@ -19,56 +19,202 @@
...
 
   /**
-   * Check if the provided digit is sensitive : 'write' bit is set to 1 (rwx)
-   * <a href="https://en.wikipedia.org/wiki/Chmod#:~:text=The%20chmod%20numerical%20format%20accepts,%2C%20setgid%2C%20and%20sticky%20flags">Wiki chmod</a>
+   * Represent chmod call instruction in RUN Arguments, with parsed permissions ready to be checked
    */
-  private static boolean isDigitRightSensitive(char digit) {
-    return ((digit - '0') & 0b10) > 0;
-  }
+  static class Chmod {
```
Comment (mstachniuk) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/PosixPermissionCheck.java:123]: I would suggest extracting this class - is quite big

File: iac-extensions/docker/src/test/resources/checks/PosixPermissionCheck/Dockerfile_run [hunk 2]
```diff
@@ -0,0 +1,204 @@
+FROM scratch
+
+## All sensitive numbers
+# Noncompliant@+1 {{Make sure this permission is safe.}}
```
Comment (mstachniuk) [inline iac-extensions/docker/src/test/resources/checks/PosixPermissionCheck/Dockerfile_run:4]: nice cases

**Review by mstachniuk — APPROVED** [review]
LGTM, see my comments

**Pull request merged**